runtime/runtime/src/lib.rs

```diff
@@ -1294,62 +1297,125 @@ impl Runtime {
             })?;
         Ok(())
     }
+}
+
+struct StorageComputer<'a> {
+    result: HashMap<String, u64>,
+    config: &'a StorageUsageConfig,
+}
+
+impl<'a> StorageComputer<'a> {
+    pub fn new(config: &'a RuntimeConfig) -> Self {
+        Self { result: HashMap::new(), config: &config.transaction_costs.storage_usage_config }
+    }
+
+    pub fn process_record<Record: Borrow<StateRecord>>(&mut self, record: Record) {
```

Comment:
```suggestion
    pub fn process_record(&mut self, record: StateRecord) {
```

I see that this is a pre-existing pattern, but it's not great. Making function generic makes compile times worse, and makes the code harder to understand. 

`Borrow` is basically only needed when your are implementing a hash map or other associative container.